hpat/hiframes/api.py

```diff
@@ -559,13 +559,13 @@ def select_k_nonan_overload(A, m, k):
     dtype = A.dtype
     if isinstance(dtype, types.Integer):
         # ints don't have nans
-        return lambda A, m, k: (A[:k].copy(), k)
+        return lambda A, m, k: (A[:max(k, 0)].copy(), k)
```

Comment:
not sure we have to cut off negatives here

Reply:
`nsmallest(0) == nsmallest(-1)` returns empty series. So we need to set up `0` instead of negative `k` to return empty array. E.g. `k = -2`, `A = [1, 2, 3, 4, 5]`:
`A[:k] # [1, 2, 3]`
`A[:0] # [] - what we want to get when we have negative k`

Reply:
BTW old functionality didn't work with negative `k`.